backend/app/admin.py

```diff
@@ -13,10 +17,118 @@
+    show_coordinates.short_description = 'Coordinates'
+
+    @staticmethod
+    def count_photos(obj):
+        """
+        Returns an integer representing the number taken of that map square
+        """
+
+        # filter for all photo objects with obj's map square number
+        photo_obj = Photo.objects.filter(map_square__number=obj.number)
+        return len(photo_obj)
+    count_photos.short_description = 'Number of Photos'
+
+    @staticmethod
+    def all_photos(obj):
+        """
+        Displays all images taken at that map square
+        """
+        cmd = ''
+        # filter for all photo objects of obj map square
+        photo_obj = Photo.objects.filter(map_square__number=obj.number)
+        for photo in photo_obj:
+            # path to photo
+            obj_path = os.path.join(settings.LOCAL_PHOTOS_DIR, str(obj.number),
+                                    str(photo.number)+'_photo.jpg')
+
+            # path to photo's admin change page
+            photo_path = os.path.join('/admin/app/photo', str(photo.id))
+            cmd += '<a target = "blank" href = "{photo_url}"> <img src="{url}" ' \
+                   'width="90px"></a>'.format(photo_url=photo_path, url=obj_path) + '\n'
+        return mark_safe(cmd)
+
+
+class PhotoAdmin(admin.ModelAdmin):
+    """
+    Photo Admin
+    """
+    list_display = ('id', 'number', 'photo_thumbnail', 'map_square_number',
+                    'photographer_info', 'librarian_caption', 'photographer_caption')
+    search_fields = ['number', 'map_square__id', 'photographer__id', 'photographer__name']
+
+    readonly_fields = ('photo_thumbnail', 'photo_slide')
+
+    def map_square_number(self, obj):
+        """
+        Returns map square number
+        """
+        link = os.path.join('/admin/app/mapsquare', str(obj.map_square.id))
+        cmd = '<a target = "blank" href = "{url}">{title}</a>'.format(url=link, title= obj.map_square.number)
+        return mark_safe(cmd)
+    map_square_number.short_description = 'Map Square'
+    map_square_number.admin_order_field = 'map_square__number'
+
+    def photographer_info(self, obj):
+        """
+        Returns photographer number (and name if available)
+        """
+        if obj.photographer:
+            if obj.photographer.name:
+                title = str(obj.photographer.id) + ' (' + obj.photographer.name + ')'
+            else:
+                title = str(obj.photographer.id) + ' (-)'
+            link = os.path.join('/admin/app/photographer', str(obj.photographer.id))
+            cmd = '<a target = "blank" href = "{url}">{title}</a>'.format(url=link, title=title)
+            return mark_safe(cmd)
+        else:
+            return
+    photographer_info.short_description = 'Photographer'
+
+    def photo_thumbnail(self, obj):
+        """
+        Displays thumbnail of photo obj
+        """
+
+        # taking photos from local dir
+        obj_path = os.path.join(settings.LOCAL_PHOTOS_DIR, str(obj.map_square.number),
```

Comment:
I think we want this to go to the THUMBNAIL_DIR (which we don't have yet). We definitely don't want to pull in all full-sized images into our admin panel